beacon-chain/db/filesystem/blob.go

```diff
@@ -102,7 +102,9 @@ func (bs *BlobStorage) Save(sidecar blocks.VerifiedROBlob) error {
 		return nil
 	}
 	if bs.pruner != nil {
-		bs.pruner.notify(sidecar.BlockRoot(), sidecar.Slot())
+		if err := bs.pruner.notify(sidecar.BlockRoot(), sidecar.Slot(), sidecar.Index); err != nil {
+			return errors.Wrapf(err, "problem maintaining pruning cache/metrics for sidecar with root=%#x", sidecar.BlockRoot())
```

Comment:
What would be the downstream effect if this returns an error?

Reply:
We wouldn't write the sidecar. The only error condition is if the index is out of bounds, which would be a panic. I was on the fence about returning an error here, because we should never have a VerifiedROBlob with an index out of bounds, so this should be logically unreachable code.

Reply:
Maybe I meant to say upstream. I think block processing will fail, but that feels sane to me. If we cant write a blob to disk then we shouldn't be allowed to write a block anyway